### PR TITLE
Fix 4.next merge

### PR DESCRIPTION
**Pull request by othercorey**

### CODE REVIEW

File: src/Error/Renderer/ConsoleExceptionRenderer.php [hunk 1]
```diff
@@ -102,11 +101,13 @@ public function render()
...
+     * @param \Psr\Http\Message\ResponseInterface|string $output The output to print.
      * @return void
      */
-    public function write($output): void
+    public function write(ResponseInterface|string $output): void
     {
-        $this->output->write($output);
+        if (is_string($output)) {
+            $this->output->write($output);
+        }
```
Comment (othercorey) [inline src/Error/Renderer/ConsoleExceptionRenderer.php:111]: @markstory We can't narrow the parameter type. Not sure what `ResponseInterface` is meant to do here.

**Comment by ADmad** [comment]
All the deprecated code seen in this diff can be dropped.

**Comment by othercorey** [comment]
I was going to do it separately. Thought this would be an easy merge but nope.

File: src/Error/Renderer/ConsoleExceptionRenderer.php (hunk 1 above)
Comment (markstory) [reply src/Error/Renderer/ConsoleExceptionRenderer.php:111]: ResponseInterface won't ever happen in a console environment. We could raise an error 🤷

Comment (ADmad) [reply src/Error/Renderer/ConsoleExceptionRenderer.php:111]: This is where using `asset()` would make sense.

**Pull request merged**